src/base/language_support.c

```diff
@@ -5522,6 +5529,11 @@ lang_fastcmp_byte (const LANG_COLLATION * lang_coll, const unsigned char *string
       return cmp;
     }
 
+  if (!ignore_trailing_space && size1 != size2)
+    {
+      return size1 - size2;
+    }
+
   c1 = c2 = ZERO;
```

Comment:
if ignore_trailing_space is set to yes then doesn't the problem occur when c1 or c2 == SPACE?
```
if (c1 == SPACE)
{
  c1 = ZERO;
}
```

Reply:
the SPACE's weight is always 0 from "weight" array, therefore the comparing result is always 0 while the c1 or c2 is SPACE.

Reply:
if so, why is it written like this below in this function?
```
      if (c2 == SPACE)
	{
	  c2 = ZERO;
	}
      else
	{
	  c2 = weight[c2];
	}
```

Reply:
in comparing step, space character 0x20 should be weight 0, it is simply for sorting regardless ignoring or not ignoring trailing space; however in comparing trailing space, the length of a string is more important than remain characters' weight if the comparing option is 'no ignoring space' regardless the weight is 0 or not.

Reply:
I thought the weight[SPACE] is not ZERO because there is the code below on line 5505 in this function.
```
        if (c1 == SPACE)
	{
	  c1 = ZERO;
	}
      else
	{
	  c1 = weight[c1];
	}
```
if weight[SPACE] is ZERO then the above code can be removed. Otherwise above code is need when ignore_trailing_space  is yes.

Reply:
you're write but when ignore_trailing_space setting is no, weight[SPACE] is not zero, so the code is needed.